src/client/components/forms/type-editor/typeUtils.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Shivam Awasthi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import type {EntityTypeString} from 'bookbrainz-data/lib/types/entity';
+import React from 'react';
+
+
+export interface RelationshipTypeDataT {
+	attributeTypes: number[];
+	id?: number;
+	label: string;
+	description: string;
+	linkPhrase: string;
+	reverseLinkPhrase: string;
+	deprecated: boolean;
+	parentId: number | null;
+	childOrder: number;
+	sourceEntityType: EntityTypeString;
+	targetEntityType: EntityTypeString;
+}
+
+export interface AttributeTypeDataT {
+	childOrder: number,
+	description: string,
+	id?: number,
+	lastUpdated: Date,
+	name: string,
+	parent: number | null,
+	root: number
+}
+
+export interface RelationshipTypeEditorPropsT {
+	relationshipTypeData: RelationshipTypeDataT;
+	parentTypes: RelationshipTypeDataT[];
+	attributeTypes: AttributeTypeDataT[];
+	user: any;
+}
+
+export const defaultRelationshipTypeData: RelationshipTypeDataT = {
+	attributeTypes: [],
+	childOrder: 0,
+	deprecated: false,
+	description: '',
+	label: '',
+	linkPhrase: '',
+	parentId: null,
+	reverseLinkPhrase: '',
+	sourceEntityType: null,
+	targetEntityType: null
+};
+
+export const entityTypeOptions = ['Author', 'Work', 'Series', 'Edition', 'EditionGroup', 'Publisher'].map((entity) => ({
```

Comment:
Indeed, I've just came across a few places in bb-data where I can also make use of this constant now that I'm refactoring the type definitions :grin:
So I will definitely create the definition over there right now.

Reply:
I think it would be best if I make the bb-data types related changes at the end. It would help me align my changes to the ones you're going to make in your (upcoming?) PR.

Reply:
Yeah, I guess so. Theoretically that part is ready, but some of the changes could be considered breaking changes and I don't want to cause a major version bump every time I have some more changes in bb-data.
It's probably better to wait with a new releases until I've finished my first pass of improving the type definitions.